hydra-tui/src/Hydra/TUI.hs

```diff
@@ -281,38 +287,38 @@ handleCommitEvent ::
   Client CardanoTx IO ->
   State ->
   EventM n (Next State)
-handleCommitEvent Client{sendInput} = \case
-  s@State{headState = Initializing{}} ->
-    case s ^. meL of
+handleCommitEvent Client{sendInput} s = case s ^? headStateL of
+  Just Initializing{} ->
+    case s ^? meL of
       -- XXX(SN): this is just..not cool
-      Nothing -> continue $ s & feedbackL ?~ UserFeedback Error "Missing identity, so can't commit from faucet."
-      Just me ->
+      Just (Just me) ->
         continue $ s & dialogStateL .~ commitDialog (faucetUtxo me)
-  s ->
+      _ -> continue $ s & feedbackL ?~ UserFeedback Error "Missing identity, so can't commit from faucet."
+  _ ->
     continue $ s & feedbackL ?~ UserFeedback Error "Invalid command."
  where
   commitDialog u =
     Dialog title form submit
    where
     title = "Select UTXO to commit"
     form = newForm (utxoCheckboxField u) ((,False) <$> u)
-    submit s selected = do
+    submit s' selected = do
       let commit = UTxO . Map.mapMaybe (\(v, p) -> if p then Just v else Nothing) $ selected
       liftIO (sendInput $ Commit commit)
-      continue (s & dialogStateL .~ NoDialog)
+      continue (s' & dialogStateL .~ NoDialog)
 
 handleNewTxEvent ::
   Client CardanoTx IO ->
   State ->
   EventM n (Next State)
-handleNewTxEvent Client{sendInput} = \case
-  s@State{headState = Open{parties}} ->
-    case s ^. meL of
+handleNewTxEvent Client{sendInput} s = case s ^? headStateL of
+  Just Open{parties} ->
```

Comment:
Question: couldn't you just compose the 2 lenses 
```
case s ^? headStateL . meL of
    Just me -> ...
    Nothing -> ...
```

Reply:
Yes, more refactoring ins possible here